src/components/Table/components/TableCell/TableCell.tsx

```diff
@@ -24,10 +27,23 @@ const TableCell: React.FC<Props> = React.memo(
     children,
     type = 'normal',
     padded = false,
+    dataTestIdPrefix,
+    rowIndex,
+    index,
   }) => {
     const theme = useTheme();
     const Component = component;
 
+    const tableCellTestId = children
+      ? component === 'th' && typeof children === 'string'
```

Comment:
we wrap the checkboxes in th, leading to inconsistencies regarding the rule you are trying to apply 
- th : **table_header_**
- td : **table_row_**


<img width="431" alt="table-th" src="https://user-images.githubusercontent.com/24905735/118146441-4b916780-b417-11eb-872d-267168ae15de.png">

Reply:
after a discussion on this, we will keep it as is it